contracts/gasToken/GasTokenForwarder.sol

```diff
@@ -0,0 +1,200 @@
+pragma solidity ^0.5.13;
+import "./IGST2.sol";
+import "../RelayerManager.sol";
+import "../libs/SafeMath.sol";
+import "../libs/Ownable.sol";
+
+contract GasTokenForwarder is Ownable {
+    using SafeMath for uint256;
+
+    IGST2 private _gasToken;
+    RelayerManager private _relayerManager;
+
+    event Efficiency(uint256 indexed efficiency, uint256 indexed tokens);
+
+    // MODIFIERS
+    modifier onlyRelayer() {
+        require(
+            _relayerManager.getRelayerStatus(msg.sender),
+            "You are not allowed to perform this operation"
+        );
+        _;
+    }
+
+    constructor(
+        address owner,
+        address gasToken2,
+        address relayerManager
+    ) public Ownable(owner) {
+        require(gasToken2 != address(0), "gasToken2 cannot be 0");
+        require(relayerManager != address(0), "Manager address cannot be 0");
+
+        _gasToken = IGST2(gasToken2);
+        _relayerManager = RelayerManager(relayerManager);
+    }
+
+    function addRelayerManager(address relayerManagerAddress) public onlyOwner {
+        require(
+            relayerManagerAddress != address(0),
+            "Manager address can not be 0"
```

Comment:
Can make strings in require statements shorter to save gas, something like "can't set to 0" or something shorter